packages/token-bridge-sdk/src/context/balanceContext.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useState,
+  PropsWithChildren
+} from 'react'
+import { BigNumber } from 'ethers'
+
+type Balance = {
+  [chainId: string]: {
+    eth: BigNumber | null
+  }
+}
+type State = {
+  [walletAddress: string]: Balance
+}
+
+type SetBalanceParams = {
+  walletAddress: string
+  chainId: number
+  type: 'eth'
+  balance: BigNumber
+}
+const BalanceContext = createContext<
+  [State, (params: SetBalanceParams) => void]
+>([{}, () => {}])
+
+function BalanceContextProvider({ children }: PropsWithChildren<{}>) {
+  const [balances, setBalances] = useState<State>({})
+
+  const setBalance = useCallback(
+    ({ walletAddress, chainId, type, balance }: SetBalanceParams) => {
+      setBalances(oldBalances => {
+        return {
+          ...oldBalances,
+          [walletAddress]: {
+            ...oldBalances[walletAddress],
+            [chainId]: {
+              ...oldBalances[walletAddress]?.[chainId],
+              [type]: balance
+            }
+          }
+        }
+      })
+    },
+    []
+  )
+
+  return (
+    <BalanceContext.Provider value={[balances, setBalance]}>
+      {children}
+    </BalanceContext.Provider>
+  )
+}
+
+function useBalanceContext() {
+  const context = useContext(BalanceContext)
+
+  if (context === undefined) {
```

Comment:
`typeof context === 'undefined'`

Reply:
Changed to be consistent with rest of the codebase.

But for information, this should still work https://stackoverflow.com/a/4725697/2551816

> For undeclared variables, typeof foo will return the string literal "undefined", whereas the identity check foo === undefined would trigger the error "foo is not defined".
> 
> For local variables (which you know are declared somewhere), no such error would occur, hence the identity check.

Reply:
For sure, but I think given that `typeof context === 'undefined'` is basically a superset of `context === undefined` we should just get used to using that always